pubtools/_quay/signature_handler.py

```diff
@@ -531,39 +492,73 @@ def sign_container_images(self, push_items, only_v2s1_manifests=False):
         - send messages to RADAS and receive signatures (ManifestClaimsHandler class)
         - Upload new signatures to Pyxis
 
-        NOTE: The reason for separating the creation of schema1 and schema2 claim messages is that
-        they must happen in different stages of the push, schema2 before pushing to destination,
-        and schema1 after pushing to destination.
+        # NOTE: The reason for separating the creation of schema1 and schema2 claim messages is that
+        # they must happen in different stages of the push, schema2 before pushing to destination,
+        # and schema1 after pushing to destination.
 
         Args:
             push_items (([ContainerPushItem])):
                 Container push items whose images will be signed.
-            only_v2s1_manifests (bool):
-                If True, only claim messages for schema1 manifests are created. If False,
-                only claim messages for schema2 manifests are created.
         """
         if not self.target_settings["docker_settings"].get(
             "docker_container_signing_enabled", False
         ):
             LOG.info("Container signing not allowed in target settings, skipping.")
-            return
+            return []
 
         claim_messages = []
         for item in push_items:
-            if only_v2s1_manifests:
-                claim_messages += self.construct_item_schema1_claim_messages(item)
-            else:
-                claim_messages += self.construct_item_claim_messages(item)
+            claim_messages += self.construct_item_claim_messages(item)
         claim_messages = self.remove_duplicate_claim_messages(claim_messages)
         claim_messages = self.filter_claim_messages(claim_messages)
         if not claim_messages:
             LOG.info("No new claim messages will be uploaded")
-            return
+            return []
+
+        LOG.info("{0} claim messages will be uploaded".format(len(claim_messages)))
+        signature_messages = self.get_signatures_from_radas(claim_messages)
+        self.validate_radas_messages(claim_messages, signature_messages)
+        self.upload_signatures_to_pyxis(claim_messages, signature_messages)
+        return signature_messages
+
+    @log_step("Sign container images")
+    def sign_container_images_new_digests(self, push_items):
+        """
+        Sign digests for manifest types which haven't exist before the push.
+
+        Manifest types which are not pushed in the push items can be fetched
```

Comment:
V2S1 was never meant to be supported by pubtools-quay but this solution permanently bakes it into its DNA. Future developers will wonder why it's here in the first place.

I wanted the migration related changes to be separated from the original code, explicitly noted that they exist just for the migration, and later removed in the future. You seem to disagree with this design. Can we discuss it?